lisp/plugins/cart_layout/page_widget.py

```diff
@@ -99,10 +101,22 @@ def contextMenuEvent(self, event):
         self.contextMenuRequested.emit(event.globalPos())
 
     def dragEnterEvent(self, event):
-        if event.mimeData().text() == CartPageWidget.DRAG_MAGIC:
+        if event.mimeData().hasUrls() and all([x.isLocalFile() for x in event.mimeData().urls()]):
             event.accept()
+        elif event.mimeData().hasText():
+            if event.mimeData().text() == CartPageWidget.DRAG_MAGIC:
+                event.accept()
+            else:
+                event.ignore()
+        else:
+            event.ignore()
 
     def dropEvent(self, event):
+        if event.mimeData().hasUrls():
+            # If files are being dropped, add them as cues
+            get_backend().cue_audio_urls(event.mimeData().urls())
+            return
+
```

Comment:
This condition change significantly the way the function behave, it would be better to use an `else` statement instead of the `return`, it will make the flow more clear.